### ISSUE TITLE
Some parts of the doc info (on hover) are incorrectly wrapped inside code tags

### ISSUE THREAD

**Issue by moodmosaic** [digital-asset/ghcide#341]
The overall feature looks neat, but some parts of the `:doc` are incorrectly wrapped in `<code></code>`

![image](https://user-images.githubusercontent.com/287532/72987401-9c2c1e00-3dea-11ea-8dd0-2137024fb873.png)

As @serras [points out](https://github.com/digital-asset/ghcide/pull/317#issuecomment-577678395), the origin of this may be at the transform of Haddock into Markdown.

---

Should this be reported in https://github.com/haskell/haddock instead, please let me know.

**Comment by serras** [comment]
This seems to be caused by the fact that the docs for `unsafePerformIO` read as follows:

```
  * Use @{\-\# NOINLINE foo \#-\}@ as a pragma on any function @foo@
        that calls 'unsafePerformIO'.  If the call is inlined,
        the I\/O may be performed more than once.
```

Unfortunately, having so many spaces before a line is parsed by Markdown as being a code block, hence the problem.

**Comment by moodmosaic** [comment]
Oh, I guess we can send them a pull request :wink: Thank you for looking into it :100:

**Comment by serras** [comment]
I really think this is on our side, since the comment in `unsafePerformIO` is valid Haddock. I've put up a fix in #344, which seems to render those docs correctly and not break anything else.
